packages/runtime-core/src/errorHandling.ts

```diff
@@ -66,18 +66,22 @@ export function callWithErrorHandling(
 }
 
 export function callWithAsyncErrorHandling(
-  fn: Function,
+  fn: Function | Function[],
   instance: ComponentInternalInstance | null,
   type: ErrorTypes,
   args?: any[]
 ) {
-  const res = callWithErrorHandling(fn, instance, type, args)
-  if (res != null && !res._isVue && isPromise(res)) {
-    res.catch((err: Error) => {
-      handleError(err, instance, type)
-    })
+  function call(handler: Function) {
```

Comment:
Good catch.

Since lifecycle hooks can be somewhat a hotspot, it's best to avoid creating a function for every call here. I'd just export a function that handles arrays but calls the original `callWithAsyncErrorHandling` instead. Also use a for loop instead of `.map` to avoid overhead, since in all the affected cases the return value doesn't matter.

Reply:
I see. I'll update later

Reply:
updated

Reply:
I think you might have to rebase from `master` There are conflicts in your branch.

Reply:
done

Reply:
You will have to do the same again. If you can give me write access to your fork, I can push these changes to your branch.

Reply:
@ayush987goyal Thanks, I've already done it